src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/AutoDrier.java

```diff
@@ -0,0 +1,134 @@
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import me.mrCookieSlime.Slimefun.api.energy.ChargableBlock;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.block.Block;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.Damageable;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AutoDrier extends AContainer {
+
+    public AutoDrier(Category category, ItemStack item, String id, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, id, recipeType, recipe);
+    }
+
+    @Override
+    public String getInventoryTitle() {
+        return "&eAuto Drier";
+    }
+
+    @Override
+    public ItemStack getProgressBar() {
+        return new ItemStack(Material.CAMPFIRE);
+    }
+
+    @Override
+    public void registerDefaultRecipes() { }
+
+    @Override
+    protected void tick(Block b) {
+        if (isProcessing(b)) {
+            int timeleft = progress.get(b);
+            if (timeleft > 0) {
+                ItemStack item = getProgressBar().clone();
+                ItemMeta im = item.getItemMeta();
+                ((Damageable) im).setDamage(MachineHelper.getDurability(item, timeleft, processing.get(b).getTicks()));
+                im.setDisplayName(" ");
+                List<String> lore = new ArrayList<>();
+                lore.add(MachineHelper.getProgress(timeleft, processing.get(b).getTicks()));
+                lore.add("");
+                lore.add(MachineHelper.getTimeLeft(timeleft / 2));
+                im.setLore(lore);
+                item.setItemMeta(im);
+
+                BlockStorage.getInventory(b).replaceExistingItem(22, item);
+
+                if (ChargableBlock.isChargable(b)) {
+                    if (ChargableBlock.getCharge(b) < getEnergyConsumption()) return;
+                    ChargableBlock.addCharge(b, -getEnergyConsumption());
+                    progress.put(b, timeleft - 1);
+                }
+                else progress.put(b, timeleft - 1);
+            }
+            else {
+                BlockStorage.getInventory(b).replaceExistingItem(22, new CustomItem(new ItemStack(Material.BLACK_STAINED_GLASS_PANE), " "));
+                pushItems(b, processing.get(b).getOutput());
+
+                progress.remove(b);
+                processing.remove(b);
+            }
+        }
+        else {
+            MachineRecipe r = null;
+            int inputSlot = -1;
+            for (int slot: getInputSlots()) {
+                ItemStack item = BlockStorage.getInventory(b).getItemInSlot(slot);
+                if (item == null) continue;
+
+                // Check if dryable
+                Material mat = item.getType();
+                boolean dryable = mat == Material.ROTTEN_FLESH || mat == Material.WATER_BUCKET || mat == Material.WET_SPONGE || Tag.SAPLINGS.isTagged(mat) || Tag.LEAVES.isTagged(mat) || mat.name().contains("POTION");

```

Comment:
This check is redundant since you do all these checks again below and continue; if they haven't been met.